### DISCUSSION TITLE
iseq ${domain}

### DISCUSSION THREAD

**Discussion by christoph-wijns** [ipxe/ipxe#562] [Q&A] [answered]
Hello and Goodmorning ,

Trying to use **domain** variable to navigate to right WDS server ...

:prd-w10
iseq ${domain} domain1.local || goto **domain1**
iseq ${domain} domain2.local || goto **domain2**
**:domain1**
imgfree
set base http://ipxe.lanservice.be/ipxe/Boot
kernel ${base}/wimboot
set wdsserver:ipv4 1.2.3.4
set netX/next-server ${wdsserver}
chain tftp://${wdsserver}/smsboot\${arch}\wdsmgfw.efi
boot
**:domain1**
imgfree
set base http://ipxe.lanservice.be/ipxe/Boot
kernel ${base}/wimboot
set wdsserver:ipv4 5.6.7.8
set netX/next-server ${wdsserver}
chain tftp://${wdsserver}/smsboot\${arch}\wdsmgfw.efi
boot

But iPXE does not like my code ;)

Already tried

iseq ${domain} domain1 || goto domain1
iseq ${domain} "domain1" || goto domain1
iseq "${domain}" "domain1" || goto domain1
iseq ${domain} "domain1" || goto domain1
iseq ${net0/domain} "domain1" || goto domain1
iseq ${net0.dhcp/domain} "domain1" || goto domain1

![2022-01-17_10h32_04](https://user-images.githubusercontent.com/75322374/149744169-36f7a54c-ea0f-4536-8340-afd33ff70714.png)

When I don't specify iseq command iPXE is booting perfectly into WDS

I any can help - much appreciated !

**Comment by NiKiZe** [comment]
What do you see when this runs, and why are you saying it isnt working?
Do you get any error message? if so what is it?

**Comment by mcb30** [comment, marked as answer]
> But iPXE does not like my code ;)

Two things:

1. Your code currently includes two labels `:domain1`: I assume that the second one is supposed to be `:domain2`.
2. Your line
    ```
    iseq ${domain} domain1 || goto domain1
    ```
    was probably supposed to mean "if `${domain}` equals `domain1` then go to label `:domain1` (otherwise fall through to the next line)", which should be written as
    ```
    iseq ${domain} domain1 && goto domain1 ||
    ```
    More efficiently, you could write just
    ```
    goto ${domain} ||
    ```
    i.e. "go to label matching domain name if such a label exists (otherwise fall through to the next line)".

HTH,

Michael

**Reply by christoph-wijns** [reply]
Thank you guys !!

This one works also 

iseq ${domain:string} domain1.local && set wdsserver:ipv4 1.2.3.4 ||
iseq ${domain:string} domain2.local && set wdsserver:ipv4 4.5.6.7 ||

Many many thanks
